### PR TITLE
test: begin implementing basic tests

### LINKED ISSUE #112: Add more complete test coverage
Just to make sure we don't break things 👍

### PR DESCRIPTION
**Pull request by andrewda**
Fixes #112.

### CODE REVIEW

**Comment by alejandronanez** [comment]
Awesome @andrewda!
What do you think about using `enzyme`? I've used it for the last year on my last project and it's a really good/solid project with a great api + backed by Airbnb which is a major player in the RN scene. https://github.com/airbnb/enzyme/blob/master/docs/guides/react-native.md

**Comment by andrewda** [comment]
That's how I want to do it @alejandronanez. Unfortunately, Enzyme does not support React v16 (until it's officially released, not in alpha) which React-Native 0.46 uses for some silly reason. See more here: https://github.com/airbnb/enzyme/issues/928

**Comment by alejandronanez** [comment]
Oh, I didn't know about that compat issue :(
We should wait then, it's better to have a good foundation from the beginning!

**Comment by andrewda** [comment]
@alejandronanez Agreed! Will keep this PR as WIP until then. React v16 should hopefully be released within the next couple weeks, and the next version of RN a few days after.

**Comment by andrewda** [comment]
React 16 and Enzyme 3 are officially released! :tada: 

I'll give Enzyme 3 a shot today. Hopefully React-Native upgrades to React 16 soon!

**Comment by alejandronanez** [comment]
@andrewda awesome!

File: package.json [hunk 1]
```diff
@@ -54,25 +56,26 @@
...
     "react-native-cookies": "^3.2.0",
     "react-native-device-info": "^0.11.0",
-    "react-native-elements": "^0.9.3",
+    "react-native-elements": "^0.17.0",
     "react-native-htmlview": "^0.12.0",
     "react-native-i18n": "^2.0.4",
     "react-native-material-design-searchbar": "^1.1.4",
-    "react-native-parallax-scroll-view": "^0.19.0",
+    "react-native-mock": "https://github.com/shqld/react-native-mock/tarball/master",
+    "react-native-parallax-scroll-view": "https://github.com/mini-eggs/react-native-parallax-scroll-view/tarball/react16-support",
```
Comment (andrewda) [inline package.json]: Definitely need to wait until these are all actually fixed before merging this. Using these forks is just to get it to work, not a permanent (or acceptable) solution.

File: package.json [hunk 2]
```diff
@@ -54,25 +56,26 @@
...
     "react-native-photo-view": "^1.5.2",
     "react-native-safari-view": "^2.0.0",
-    "react-native-search-bar": "^3.0.0",
+    "react-native-search-bar": "https://github.com/andrewda/react-native-search-bar/tarball/master",
```
Comment (andrewda) [inline package.json]: Same as above

File: package.json [hunk 3]
```diff
@@ -54,25 +56,26 @@
...
     "react-native-syntax-highlighter": "^1.2.1",
-    "react-native-table-component": "^1.0.5",
+    "react-native-table-component": "https://github.com/jaynakus/react-native-table-component/tarball/master",
```
Comment (andrewda) [inline package.json]: Same as above

**Comment by andrewda** [comment]
For now I've only went ahead and tested `<CodeLine />`. More to come.

**Comment by andrewda** [comment]
`[missing "en.common.info" translation]` when rendering components with translations - not sure how to fix this.

**Comment by andrewda** [comment]
Moved to #407 to make contributions easier.

**Pull request closed without merging**